sdk/storage/storage-file-share/src/Clients.ts

```diff
@@ -2756,6 +2760,78 @@ export class ShareDirectoryClient extends StorageClient {
       span.end();
     }
   }
+
+  /**
+   * Renames a directory.
+   * This API does not support renaming a directory from one share to another, or between storage accounts.
+   *
+   * @param destinationPath - Specifies the destination path to rename to.
+   * @param options - Options for the renaming operation.
+   * @returns Response data for the file renaming operation.
+   *
+   * Example usage:
+   *
+   * ```js
+   *
+   * // Rename the directory
+   * await diretoryClient.rename(destinationPath);
+   * console.log("Renamed directory successfully!");
+   * ```
+   */
+  public async rename(
+    destinationPath: string,
+    options: DirectoryRenameOptions = {}
+  ): Promise<{
+    destinationDirectoryClient: ShareDirectoryClient;
+    directoryRenameResponse: DirectoryRenameResponse;
+  }> {
+    const { span, updatedOptions } = createSpan("ShareDirectoryClient-rename", options);
+    const split: string[] = destinationPath.split("?");
+    let destinationUrl: string;
+    if (split.length === 2) {
+      const pathOnly = encodeURIComponent(split[0]);
```

Comment:
Is that possible we double encoded the path? And why we only URI encode the path part?

Reply:
destinationPath can include two part: 1. the actual path includes directory and file name 2. query string which could be a SAS token. 

Only actual path needed to be encoded, we should not encode SAS token for it.

Reply:
May elaborate the comment.